src/kudu/consensus/raft_consensus.cc

```diff
@@ -193,6 +193,13 @@ DEFINE_bool(raft_enforce_rpc_token, false,
             "Should enforce that requests and reponses to this instance must "
             "have a matching token as what we have stored.");
 
+DEFINE_bool(withhold_votes_on_huge_lag, false,
+            "Should a voter withhold their votes to CANDIDATE, if they lag "
+            "significantly behind");
+
+DEFINE_int32(huge_lag_threshold_for_request_vote, 3600000,
```

Comment:
This is 3600000 opids of lag? Isn't that a lot?

Reply:
Just call it lag_threshold_for_request_vote maybe?

Reply:
this is 3.6M trxs which at the rate of 20K trxs per second (upper limit) is 180 seconds of lag. This is almost the
same calculation we use to give up on catchups.

Reply:
Hmm, it would be best if this threshold is based on either time or space, num opids is not the perfect metric for this IMO. However, i think it'll be very hard to make this time or space based unless we have the gossip framework. Maybe just add a TODO here for now.